rapids-cmake/cuda/set_architectures.cmake

```diff
@@ -56,14 +56,18 @@ Result Variables
 function(rapids_cuda_set_architectures mode)
   list(APPEND CMAKE_MESSAGE_CONTEXT "rapids.cuda.set_architectures")
 
-  set(supported_archs "70" "75" "80" "86" "90")
+  set(supported_archs "70" "75" "80" "86" "90" "100" "120")
 
   if(CMAKE_CUDA_COMPILER_ID STREQUAL "NVIDIA" AND CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 11.1.0)
     list(REMOVE_ITEM supported_archs "86")
   endif()
   if(CMAKE_CUDA_COMPILER_ID STREQUAL "NVIDIA" AND CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 11.8.0)
     list(REMOVE_ITEM supported_archs "90")
   endif()
+  if(CMAKE_CUDA_COMPILER_ID STREQUAL "NVIDIA" AND CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 12.8.0)
+    list(REMOVE_ITEM supported_archs "120")
+    list(REMOVE_ITEM supported_archs "100")
+  endif()
```

Comment:
Given this removes the new architectures if the CUDA version is less than 12.8.0, this seems safe to add now